cogent/src/Cogent/C/Syntax.hs

```diff
@@ -174,3 +176,45 @@ data StrlType = Record  [(CId, CType)]       -- ^ @(fieldname &#x21A6; fieldtype
               | Array CType (Maybe Int)
               deriving (Eq, Ord, Show)
 
+
```

Comment:
Shall we define equality directly on `StrlType`? Is `StrlCogentType` solely for defining a customised equality?

Reply:
It’s solely for customised equality. You could do that, but then there is a lot more code you have to write which is prone to errors - this way, you can use the auto generated Eq instance for StrlType and only have to deal with writing the Eq instance of the BoxedRecord case.